dhis-2/dhis-api/src/main/java/org/hisp/dhis/icon/CustomIconService.java

```diff
@@ -63,14 +62,23 @@ public interface IconService {
    */
   CustomIcon getCustomIcon(String key) throws NotFoundException;
 
+  /**
+   * Gets the custom icon associated to a uid, if it exists
+   *
+   * @param uid key of the icon to find
+   * @return custom icon associated to the uid, if found
+   * @throws NotFoundException if no custom icon exists with the provided uid
+   */
+  CustomIcon getCustomIconByUid(String uid) throws NotFoundException;
```

Comment:
I thought we were clear on icons not having a UID as it collides with the `key`.

Reply:
i think we should keep it if we plan to make custom icon a metadata object and for sake of consistency as well.

Reply:
Hm... my last understanding was that icons are a bad fit for classic metadata and making them look like metadata is more pain than gain.

Reply:
fixed